readthedocs/oauth/services/base.py

```diff
@@ -181,11 +182,30 @@ def paginate(self, url, **kwargs):
                 url,
                 debug_data,
             )
-            return []
+
+        return []
 
     def sync(self):
-        """Sync repositories and organizations."""
-        raise NotImplementedError
+        """
+        Sync repositories (RemoteRepository) and organizations (RemoteOrganization).
+
+        - creates a new RemoteRepository/Organization per new repository
+        - updates fields for existing RemoteRepository/Organization
+        - deletes old RemoteRepository/Organization that are not present for this user
+        """
+        repos = self.sync_repositories()
+        organizations, organization_repos = self.sync_organizations()
+
+        # Delete RemoteRepository where the user doesn't have access anymore
```

Comment:
I wonder if we should run these delete tasks across all users as a one-time CLI command, to clean up our DB. There's likely a ton of these sitting around.

Reply:
This one-time command would require to sync _all VCS_ account connected and delete all the old objects. That may be too much. We may want to limit the amount of accounts re-synced and run it several times by chunks. It will remove a lot of stale data.

However, as this cleanup is in the core of the sync action, all active accounts will be cleanup automatically when they login next time or, when they click the arrow from Import Project page.